sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -89,6 +89,22 @@ object Cast {
     case _ => false
   }
 
+  def needsTimeZone(from: DataType, to: DataType): Boolean = (from, to) match {
```

Comment:
Nit: private? 

Could you write comments to explain why we only consider these type pairs?